### ISSUE TITLE
DS data page: No error messages for empty fields and deletion when adding DS data record 3 and beyond

### ISSUE THREAD

**Issue by AnnaGingle** [cisagov/manage.get.gov#2774]
### Issue description

We need to validate that this is still an issue with the newly redesigned DS Data page. 

- [ ] If this isn't still an issue, this can be closed as wont-do. 
- [ ] If it's still an issue, do it if it can be completed as a 1. Create a new ticket if it's larger. 

### Current Behavior

When a user attempts to add a 3rd DS data record (or any subsequent record after 3) but does not fill out the required fields for DS data record 3 and then clicks "Save," the form automatically deletes DS data record 3 without displaying any error messages. 

This behavior violates Section 508 Trusted Tester IDs [3.3.1-error-identification](https://section508coordinators.github.io/TrustedTester/forms.html) (WCAG [3.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-identified.html)) and [3.3.4-error-prevention](https://section508coordinators.github.io/TrustedTester/forms.html) (WCAG [3.3.4](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-reversible.html)).

https://github.com/user-attachments/assets/0a9c9a4a-1242-4ab8-819d-f7a2f7d0eb46

### Expected Behavior

When a user creates a new record and they did not fill out the required fields. The user presses save and they are thrown errors prompting the user to fill out the required fields.
![getgov-staging app cloud gov_domain_9950_dns_dnssec_dsdata (1)](https://github.com/user-attachments/assets/80c17ae6-dfef-4c9a-a792-5099a9684040)

### Steps to Reproduce

1. Create and fill out all the required fields for 2 DS data records
2. Create a third DS data record and do not fill out any fields for that record
3. Press save

### Environment

Environment: Staging
Browser: Chrome

### Issue Links

- ⏬ Caused by DS data page cancel button resets form #2700 
- 🔄 Relates to DS data page: Associate delete button with according DS record #2773 
- 🚫 This shouldn't be worked on until #2911 and #2368 are complete, since they may resolve this issue.

### Prioritization Comments

**WCAG Conformance Level:**  2.1 AA
**WCAG Standard that applies:** 
3.3.1 - Error Identification (Critical)
3.3.4 - Error Prevention (Critical, partially supports)
**Critical Failure if Not Met?:**  
**Prioritization:** High/Must Do
**Reason:** This issue impacts both user experience and accessibility by failing to inform users of required field errors before automatically deleting their input. This can lead to user confusion and frustration, as well as a loss of entered data without warning, which is especially problematic for users with disabilities who rely on error prompts for navigation and form completion.

**Comment by AnnaGingle** [comment]
[Designs](https://www.figma.com/design/xFtGLHVrhp0lvwh0gYWnbx/.gov-registar?node-id=9244-22349&t=JkMWw7KF0iFTAmDT-4) are ready for review

**Comment by gabydisarli** [comment]
Note: this may resolve itself once the updates in #2911 are implemented. @abroddrick

**Comment by h-m-f-t** [comment]
This is slotted for 61 (Jan 6), but our portended fix is in 62...

**Comment by PaulKuykendall** [comment]
> This is slotted for 61 (Jan 6), but our portended fix is in 62...

We'll look at this in Sprint Planning today.

**Comment by PaulKuykendall** [comment]
Move to Sprint 63, because 2911 might resolve this

**Comment by erinysong** [comment]
Tested on manage.get.gov that this bug still exists - adding and saving a 3rd DS record without any fields will just remove that 3rd DS record. The errors only populate when at least 1 but not all DS record field is filled. I try fixing this so that added DS records with empty fields are preserved with errors

**Comment by erinysong** [comment]
Tested sandbox with [PR 1208](https://github.com/cisagov/manage.get.gov/pull/3630) and we still get no errors when submitting an empty DS Data form. Given the new DS data format, do we still want to display errors when submitting a completely empty DS data form, as opposed to our current approach which just removes any empty DS Data forms that are saved?

**Comment by katypies** [comment]
Adding a link to a [Slack discussion](https://cisa-corp.slack.com/archives/C03QM0JGSQG/p1742928998817119) with some additional context here. We're going to hold off on this for right now, and de-prioritize in favor of some other in-progress items that need to be completed for end-of-POP.
